src/prefect/tasks/databricks/models.py

```diff
@@ -209,3 +209,28 @@ class AccessControlRequestForGroup(BaseModel):
 
 class AccessControlRequest(BaseModel):
     __root__: Union[AccessControlRequestForUser, AccessControlRequestForGroup]
+
+class GitSource(BaseModel):
+    git_url: str
+    git_provider: str
+    git_branch: Optional[str] = None
+    git_tag: Optional[str] = None
+    git_commit: Optional[str] = None
+
+    @validator('git_tag', always=True)
+    def branch_or_tag(cls, v, values):
+        if 'git_branch' in values is not None and v:
+            raise ValueError('Cannot specify git tag if git branch has been specified')
+        return v
+
+    @validator('git_branch', always=True)
+    def commit_or_branch(cls, v, values):
+        if 'git_commit' in values is not None and v:
+            raise ValueError('Cannot specify git branch if git commit has been specified')
+        return v
+
+    @validator('git_commit', always=True)
+    def tag_or_commit(cls, v, values):
+        if 'git_tag' in values is not None and v:
+            raise ValueError('Cannot specify git commit if git tag has been specified')
+        return v
```

Comment:
minor linting nitpick; enter new line at end of file
```suggestion
        return v

```